rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/profile/AlarmRuleState.java

```diff
@@ -467,6 +470,10 @@ private boolean evalStrPredicate(DataSnapshot data, EntityKeyValue ekv, StringFi
                 return !val.equals(predicateValue);
             case NOT_CONTAINS:
                 return !val.contains(predicateValue);
+            case IN:
+                return equalsAny(val, getListValuesWithoutQuote(predicateValue).toArray(new String[0]));
```

Comment:
Let's implement our own `equalsAny` that takes a list instead of array, to avoid `toArray(new String[0])`